### PR TITLE
MAUI-2359-UGImages-Recurrence helper method added.

### PR DESCRIPTION
**Pull request by Indumathi1195R**
Recurrence helper method added.

Content review task: https://syncfusion.atlassian.net/browse/DOC-4560

### CODE REVIEW

File: MAUI/Scheduler/appointments.md [hunk 1]
```diff
@@ -506,6 +506,49 @@ var date1 = 12/14/2021 10:00:00 AM;
...
+
+{% tabs %}
+{% highlight c# tabtitle="MainPage.xaml.cs" hl_lines="3" %}
+
+var dateTime = new DateTime(2022,07,22,9,0,0);
+var dateCollection = SchedulerRecurrenceManager.GetOccurrenceAppointment(Scheduler, appointment, dateTime);
+
+{% endhighlight %}
+{% endtabs%}
+
```
Comment (Jeyasri-Murugan) [inline MAUI/Scheduler/appointments.md:521]: Add method of ParseRule

File: MAUI/Scheduler/appointments.md [hunk 2]
```diff
@@ -506,6 +506,49 @@ var date1 = 12/14/2021 10:00:00 AM;
...
 
+#### How to get the occurence appointment from RRULE?
+
+The [GetOccurenceAppointment](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.Scheduler.SchedulerRecurrenceManager.html#Syncfusion_Maui_Scheduler_SchedulerRecurrenceManager_GetDateTimeOccurrences_System_String_System_DateTime_System_Nullable_System_DateTime__System_Nullable_System_DateTime__System_Nullable_System_DateTime__) method of `.NET MAUI Scheduler` returns the occurrence appointment for the given pattern appointment at the specified date.
+
+{% tabs %}
+{% highlight c# tabtitle="MainPage.xaml.cs" hl_lines="3" %}
+
+var dateTime = new DateTime(2022,07,22,9,0,0);
+var dateCollection = SchedulerRecurrenceManager.GetOccurrenceAppointment(Scheduler, appointment, dateTime);
```
Comment (Jeyasri-Murugan) [inline MAUI/Scheduler/appointments.md:517]: var dateCollection = rename as var occurrenceAppointment

File: MAUI/Scheduler/appointments.md [hunk 3]
```diff
@@ -506,6 +506,49 @@ var date1 = 12/14/2021 10:00:00 AM;
...
+
+#### How to get the pattern appointment from RRULE? 
+
+The [GetOccurenceAppointment](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.Scheduler.SchedulerRecurrenceManager.html#Syncfusion_Maui_Scheduler_SchedulerRecurrenceManager_GetDateTimeOccurrences_System_String_System_DateTime_System_Nullable_System_DateTime__System_Nullable_System_DateTime__System_Nullable_System_DateTime__) method of `.NET MAUI Scheduler` returns returns the Pattern appointment for the provided occurrence appointment.
+
+{% tabs %}
+{% highlight c# tabtitle="MainPage.xaml.cs" hl_lines="2" %}
+
+var dateTime = new DateTime(2022,07,22,9,0,0);
+var dateCollection = SchedulerRecurrenceManager.GetPatternAppointment(Scheduler,appointment);
```
Comment (Jeyasri-Murugan) [inline MAUI/Scheduler/appointments.md:530]: var dateCollection = rename as var patternAppointment

File: MAUI/Scheduler/appointments.md [hunk 4]
```diff
@@ -506,6 +506,49 @@ var date1 = 12/14/2021 10:00:00 AM;
...
+{% tabs %}
+{% highlight c# tabtitle="MainPage.xaml.cs" hl_lines="7" %}
+
+var recurrenceProperties = new SchedulerRecurrenceInfo();
+recurrenceProperties.RecurrenceType = SchedulerRecurrenceType.Daily;
+recurrenceProperties.Interval = 2;
+recurrenceProperties.RecurrenceCount = 3;
+var startTime = DateTime.Today.AddHours(9);
+var endTime = DateTime.Today.AddHours(10);
+var dateCollection = SchedulerRecurrenceManager.GenerateRRule(recurrenceProperties, startTime, endTime);
```
Comment (Jeyasri-Murugan) [inline MAUI/Scheduler/appointments.md:547]: var dateCollection = rename as var recurrenceRule

File: MAUI/Scheduler/appointments.md (hunk 1 above)
Comment (Indumathi1195R) [reply MAUI/Scheduler/appointments.md:521]: Already ParseRRule helper method added. https://help.syncfusion.com/maui/scheduler/appointments#how-to-get-the-recurrence-editor-field-values-from-rrule

File: MAUI/Scheduler/appointments.md (hunk 2 above)
Comment (Indumathi1195R) [reply MAUI/Scheduler/appointments.md]: Renamed as per the suggestions.

File: MAUI/Scheduler/appointments.md (hunk 3 above)
Comment (Indumathi1195R) [reply MAUI/Scheduler/appointments.md]: Renamed as per the suggestions.

File: MAUI/Scheduler/appointments.md (hunk 4 above)
Comment (Indumathi1195R) [reply MAUI/Scheduler/appointments.md]: Renamed as per the suggestions.

**Review by Nijamudeen-Mohamed — APPROVED** [review]
Fine

File: MAUI/Scheduler/appointments.md [hunk 5]
```diff
@@ -506,6 +506,49 @@ var date1 = 12/14/2021 10:00:00 AM;
 var date2 = 12/15/2021 10:00:00 AM;
 var date3 = 12/16/2021 10:00:00 AM;
 
+#### How to get the occurence appointment from RRULE?
+
+The [GetOccurenceAppointment](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.Scheduler.SchedulerRecurrenceManager.html#Syncfusion_Maui_Scheduler_SchedulerRecurrenceManager_GetDateTimeOccurrences_System_String_System_DateTime_System_Nullable_System_DateTime__System_Nullable_System_DateTime__System_Nullable_System_DateTime__) method of `.NET MAUI Scheduler` returns the occurrence appointment for the given pattern appointment at the specified date.
```
Comment (Chrispineagunja) [inline MAUI/Scheduler/appointments.md:511]: The [GetOccurenceAppointment](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.Scheduler.SchedulerRecurrenceManager.html#Syncfusion_Maui_Scheduler_SchedulerRecurrenceManager_GetDateTimeOccurrences_System_String_System_DateTime_System_Nullable_System_DateTime__System_Nullable_System_DateTime__System_Nullable_System_DateTime__) method of the `.NET MAUI Scheduler` returns the occurrence appointment for the given pattern appointment at the specified date.

File: MAUI/Scheduler/appointments.md [hunk 6]
```diff
@@ -506,6 +506,49 @@ var date1 = 12/14/2021 10:00:00 AM;
...
+
+var dateTime = new DateTime(2022,07,22,9,0,0);
+var occurrenceAppointment = SchedulerRecurrenceManager.GetOccurrenceAppointment(Scheduler, appointment, dateTime);
+
+{% endhighlight %}
+{% endtabs%}
+
+#### How to get the pattern appointment from RRULE? 
+
+The [GetOccurenceAppointment](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.Scheduler.SchedulerRecurrenceManager.html#Syncfusion_Maui_Scheduler_SchedulerRecurrenceManager_GetDateTimeOccurrences_System_String_System_DateTime_System_Nullable_System_DateTime__System_Nullable_System_DateTime__System_Nullable_System_DateTime__) method of `.NET MAUI Scheduler` returns returns the Pattern appointment for the provided occurrence appointment.
```
Comment (Chrispineagunja) [inline MAUI/Scheduler/appointments.md:524]: The [GetOccurenceAppointment](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.Scheduler.SchedulerRecurrenceManager.html#Syncfusion_Maui_Scheduler_SchedulerRecurrenceManager_GetDateTimeOccurrences_System_String_System_DateTime_System_Nullable_System_DateTime__System_Nullable_System_DateTime__System_Nullable_System_DateTime__) method of the `.NET MAUI Scheduler` returns the pattern appointment for the provided occurrence appointment.

File: MAUI/Scheduler/appointments.md [hunk 7]
```diff
@@ -506,6 +506,49 @@ var date1 = 12/14/2021 10:00:00 AM;
...
+{% highlight c# tabtitle="MainPage.xaml.cs" hl_lines="2" %}
+
+var dateTime = new DateTime(2022,07,22,9,0,0);
+var patternAppointment = SchedulerRecurrenceManager.GetPatternAppointment(Scheduler,appointment);
+
+{% endhighlight %}
+{% endtabs%}
+
+#### How to generate RRule from given appointment?
+The [GenerateRRule](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.Scheduler.SchedulerRecurrenceManager.html#Syncfusion_Maui_Scheduler_SchedulerRecurrenceManager_GetDateTimeOccurrences_System_String_System_DateTime_System_Nullable_System_DateTime__System_Nullable_System_DateTime__System_Nullable_System_DateTime__) method of `.NET MAUI Scheduler` Method to generates the recurrence rule based on the given recurrence properties, the start date and end date of the recurrence appointments.
```
Comment (Chrispineagunja) [inline MAUI/Scheduler/appointments.md:536]: The [GenerateRRule](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.Scheduler.SchedulerRecurrenceManager.html#Syncfusion_Maui_Scheduler_SchedulerRecurrenceManager_GetDateTimeOccurrences_System_String_System_DateTime_System_Nullable_System_DateTime__System_Nullable_System_DateTime__System_Nullable_System_DateTime__) method of the `.NET MAUI Scheduler` generates the recurrence rule based on the given recurrence properties, the start date, and the end date of the recurrence appointments.

**Comment by Indumathi1195R** [comment]
rebuild

File: MAUI/Scheduler/appointments.md [hunk 8]
```diff
@@ -506,6 +506,49 @@ var date1 = 12/14/2021 10:00:00 AM;
 var date2 = 12/15/2021 10:00:00 AM;
 var date3 = 12/16/2021 10:00:00 AM;
 
+#### How to get the occurence appointment from RRULE?
```
Comment (Chrispineagunja) [inline MAUI/Scheduler/appointments.md:509]: #### How to get the occurrence appointment from the RRULE

File: MAUI/Scheduler/appointments.md [hunk 9]
```diff
@@ -506,6 +506,49 @@ var date1 = 12/14/2021 10:00:00 AM;
...
+{% tabs %}
+{% highlight c# tabtitle="MainPage.xaml.cs" hl_lines="3" %}
+
+var dateTime = new DateTime(2022,07,22,9,0,0);
+var occurrenceAppointment = SchedulerRecurrenceManager.GetOccurrenceAppointment(Scheduler, appointment, dateTime);
+
+{% endhighlight %}
+{% endtabs%}
+
+#### How to get the pattern appointment from RRULE?
```
Comment (Chrispineagunja) [inline MAUI/Scheduler/appointments.md:522]: #### How to get the pattern appointment from the RRULE

File: MAUI/Scheduler/appointments.md [hunk 10]
```diff
@@ -506,6 +506,49 @@ var date1 = 12/14/2021 10:00:00 AM;
...
+{% tabs %}
+{% highlight c# tabtitle="MainPage.xaml.cs" hl_lines="2" %}
+
+var dateTime = new DateTime(2022,07,22,9,0,0);
+var patternAppointment = SchedulerRecurrenceManager.GetPatternAppointment(Scheduler,appointment);
+
+{% endhighlight %}
+{% endtabs%}
+
+#### How to generate RRule from given appointment?
```
Comment (Chrispineagunja) [inline MAUI/Scheduler/appointments.md:535]: #### How to generate the RRule from the given appointment

File: MAUI/Scheduler/appointments.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (Chrispineagunja) [reply MAUI/Scheduler/appointments.md:509]: How to get the occurrence appointment from the recurring appointment

How to get the recurring pattern appointment of the occurrence appointment

How to generate the RRule for the recurring appointments

**Pull request merged**